internal/serve/httphandler/profile_handler.go

```diff
@@ -200,6 +206,55 @@ func (h ProfileHandler) PatchUserProfile(rw http.ResponseWriter, req *http.Reque
 	httpjson.RenderStatus(rw, http.StatusOK, map[string]string{"message": "user profile updated successfully"}, httpjson.JSON)
 }
 
+func (h ProfileHandler) PatchUserPassword(rw http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+
+	token, ok := ctx.Value(middleware.TokenContextKey).(string)
+	if !ok {
+		httperror.Unauthorized("", nil, nil).Render(rw)
+		return
+	}
+
+	var reqBody PatchUserPasswordRequest
+	if err := httpdecode.DecodeJSON(req, &reqBody); err != nil {
+		err = fmt.Errorf("decoding the request body: %w", err)
+		log.Ctx(ctx).Error(err)
+		httperror.BadRequest("", err, nil).Render(rw)
+		return
+	}
+
+	// basic incoming parameters validation
+	v := validators.NewValidator()
+	v.Check(reqBody.CurrentPassword != "", "current_password", "current_password is required")
+	v.Check(reqBody.CurrentPassword != reqBody.NewPassword, "new_password", "new_password should be different from current_password")
+	if v.HasErrors() {
+		httperror.BadRequest("", nil, v.Errors).Render(rw)
+		return
+	}
+
+	// validate if the password format attends the requirements
+	badRequestExtras := map[string]interface{}{}
+	var validatePasswordError *authUtils.ValidatePasswordError
+	err := authUtils.ValidatePassword(reqBody.NewPassword)
+	if err != nil && errors.As(err, &validatePasswordError) {
```

Comment:
Good point! Not right now, but since the implementation could be changed in the future, we should make sure to address that issue as well.

Updated in https://github.com/stellar/stellar-disbursement-platform-backend/pull/22/commits/09b651e44d6d2cbc9728bc3b80ff24939bf7cef7, so errors of a different type will trigger a 500.